extensions/amp-cxense/0.1/amp-cxense.js

```diff
@@ -0,0 +1,276 @@
+                    self._loaded = true;
+                    return resolve(arg);
+                };
+
+                if (self._target && self._CXV) {
+                    self._CXV.Widgets.get('#' + self._id, function (embed) {
+                        self._embed = embed;
+
+                        if (self._isPlayer) {
+                            let selector = '#' + self._id + '.metaplayer';
+                            self._CXV.Widgets.get(selector, function (mpf) {
+
+                                self._mpf = mpf;
+                                mpf.listen('ready', () => {
+                                    self.applyFillContent(self._target);
+                                    newResolve(self);
+                                });
+                            });
+                        } else {
+                            self.applyFillContent(self._target);
+                            newResolve(self);
+                        }
+                    });
+                } else {
+                    newResolve(self);
+                }
+            });
+        });
+    }
+
+    /** @override */
+    unlayoutCallback() {
+        this._embed && this._embed.destroy();
+        this._target && this._target.parentNode.removeChild(this._target);
+        this._target = null;
+        return true;
+    }
+
+    /** @override */
+    pauseCallback() {
+        if (! this._pauseMpfSync()) {
+            this._pauseMpf();
+        }
+    }
+
+    /** @private */
+    _pauseMpf() {
+        let self = this;
+
+        return new Promise((resolve) => {
+            if (self._target && self._CXV) {
+                self._CXV.Widgets.get('#' + self._target.getAttribute('id') + '.metaplayer', (mpf) => {
+                    self._mpf = self._mpf || mpf;
+                    mpf.video.pause();
+                    resolve();
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    /** @private */
+    _pauseMpfSync() {
+        return !!(this._mpf && (this._mpf.video.pause() || this._mpf.video.paused));
+    }
+
+    /** @private */
+    _buildWidgetPlaceholder() {
+        const doc = this._getDoc();
+
+        let placeholder = doc.createElement('div');
+        placeholder.className = 'amp-cxense-placeholder';
+
+        let spinner = doc.createElement('div');
+        spinner.className = 'amp-mpf-loader';
+        spinner.appendChild(doc.createElement('div'));
+        spinner.appendChild(doc.createElement('div'));
+        spinner.appendChild(doc.createElement('div'));
+
+        placeholder.appendChild(spinner);
+
+        this.element.appendChild(placeholder);
+        this.applyFillContent(placeholder);
+    }
+
+    /** @private */
+    _injectEmbedScript () {
+        return this._injectScript(this._src);
+    }
+
+    /** @private */
+    _injectScript (src) {
+        const doc = this._getDoc();
+        const script = doc.createElement('script');
```

Comment:
@mkhatib, clear something out for me. Are you saying I can't (or shouldn't) execute JS at the top level window? and which I cannot add any property to `window.top` am I right?

Because, it seems like, in the Facebook Example, that the SDK is being fetched,  however, it is still executed in an iframe, under a different domain than the main page.